searchapp/views.py

```diff
@@ -402,12 +405,18 @@ def get_generic_paper(request):
 
 
 def get_customize_paper(request):
+    if not request.user.is_authenticated:
+        return JsonResponse({"error": "To access customised paper, you need to be logged in"}, status=401)
     if request.method == 'POST':
         subject = request.POST['subject']
         # filter returns a query set which is converted to a list and then the first element is picked up.
         subject_name = list(Subject.objects.filter(
             id=subject).values_list('subject_name', flat=True))[0]
-        chapters = map(int, request.POST.getlist('chapters[]')[0].split(','))
+        chapters = request.POST.getlist('chapters[]')[0].split(',')
```

Comment:
@KIRA009 why was this part changed?

Reply:
I mentioned the reason in the commit message
> when chapters list is evaluated, because of the
nature of split function, an empty string is converted to a list with an
element which is an empty string as well. Running map on this list with
int function means that int('') is run, which raises an Exception.